drivers/gpio/gpio_stellaris.c

```diff
@@ -81,7 +84,7 @@ static int gpio_stellaris_configure(struct device *dev, int access_op,
 	u32_t port_map = cfg->port_map;
 
 	/* Check for an invalid pin configuration */
-	if ((flags & GPIO_INT) && (flags & GPIO_DIR_OUT)) {
+	if (((flags & GPIO_INT_ENABLE) != 0) && ((flags & GPIO_OUTPUT) != 0)) {
```

Comment:
This check should actually go into `gpio_stellaris_pin_interrupt_configure` function. Since the `GPIO_OUTPUT` is not available there we need to read the value of `GPIO_REG_ADDR(base, GPIO_DIR_OFFSET)` register.

Reply:
Going to just remove this as nothing in the hardware precludes this.